### PR TITLE
ci: send email on blog post

### PR DESCRIPTION
**Pull request by juliodialpad**
# Send email on blog post

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [x] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [ ] Other (chore)

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1480

## :book: Description

Testing email sending with my email account to @braddialpad and myself.
once we have the credentials to the right email I'll update this PR.

## :bulb: Context

We need to send emails on every dialtone blog post

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [ ] I have added all relevant documentation.
- [ ] I have considered the performance impact of my change.

## :crystal_ball: Next Steps

- Update credentials
- Automate on blog post

### CODE REVIEW

File: .github/workflows/send-email.yml
Comment (braddialpad) [inline .github/workflows/send-email.yml]: I wonder if it will handle this md file correctly... hmm guess we'll find out.

**Review by braddialpad — APPROVED** [review]

**Comment by braddialpad** [comment]
Screenshot 2024-02-27 at 11 54 48 AM

Looks pretty good! this heading and posted thing at the top is kinda weird, but the rest looks okay. We'll probably also want to include a link to the actual post on the docsite.

We should test if images render correctly. Also codeblocks.

Seems we sometimes render vue components too, like in 2022-12-9.md. I doubt those are going to work...

File: .github/workflows/send-email.yml
Comment (juliodialpad) [reply .github/workflows/send-email.yml]: Yes, it should as it has an option to do it.

**Comment by juliodialpad** [comment]
> Screenshot 2024-02-27 at 11 54 48 AM
> 
> Looks pretty good! this heading and posted thing at the top is kinda weird, but the rest looks okay. We'll probably also want to include a link to the actual post on the docsite.
> 
> We should test if images render correctly. Also codeblocks.
> 
> Seems we sometimes render vue components too, like in 2022-12-9.md. I doubt those are going to work...

That's the frontmatter that we use for VuePress...

```md
---
heading: Release of our new blog!
author: Brad Paugh
posted: '2022-11-18'
---
```

**Comment by braddialpad** [comment]
> That's the frontmatter that we use for VuePress...

Yeah guess it wouldn't understand frontmatter..

**Comment by juliodialpad** [comment]
This was just a test and it went pretty good, just included a file to test markdown rendering.

There should be a lot of different ways to send this communication, maybe we might just want to extract the first lines of the post like the title and put the link if you want to read the full article.

We don't need to render everything on the email.

**Comment by braddialpad** [comment]
> maybe we might just want to extract the first lines of the post like the title and put the link if you want to read the full article.
> 
> We don't need to render everything on the email.

I agree it's unnecessary and email is too limited to display the full article. Maybe heading and first (text-only) paragraph.

**Comment by juliodialpad** [comment]
The action is complete, I tested a bunch of times, please give it a review.

Do we want to create a blog post to announce that we're going to start sending communications after this is done or should I create it on this PR before merge so we can send our first automated message right on?

**Comment by braddialpad** [comment]
> Do we want to create a blog post to announce that we're going to start sending communications after this is done or should I create it on this PR before merge so we can send our first automated message right on?

This is a good idea, let's do it after we merge.

**Comment by braddialpad** [comment]
Code looks fine, but can you send me the final email so I can see the format?

**Comment by juliodialpad** [comment]
> Code looks fine, but can you send me the final email so I can see the format?

Sure, I'll send an email and then update the emails

File: .github/workflows/send-email.yml
Comment (nickfirespotter) [inline .github/workflows/send-email.yml:81]: what is this action? Has it been reviewed by security team?

File: .github/workflows/send-email.yml
Comment (nickfirespotter) [inline .github/workflows/send-email.yml]: this is not a good solution. It is tying itself to your personal username and pw.
I would suggest using AWS SES for this. No idea if there is already an action for that or not.

**Review by nickfirespotter — CHANGES_REQUESTED** [review]

File: .github/workflows/send-email.yml
Comment (nickfirespotter) [reply .github/workflows/send-email.yml]: Lets consider https://github.com/marketplace/actions/action-to-send-email-via-aws-ses-without-using-smtp-credentials

We should have the GHA be reviewed by security team to ensure its not doing nefarious actions with our SES account or content we intend to send.

File: .github/workflows/send-email.yml
Comment (juliodialpad) [reply .github/workflows/send-email.yml:81]: No, it hasn't been reviewed by the security team. I didn't knew we needed to send them to review before using it.

File: .github/workflows/send-email.yml
Comment (juliodialpad) [reply .github/workflows/send-email.yml]: Yeah, that sounds like a better solution, I asked IT several days ago and they haven't replied to my message, so we went with this solution

**Comment by iropolo** [comment]
> > Code looks fine, but can you send me the final email so I can see the format?
> 
> Sure, I'll send an email and then update the emails

Could you send me an email also please.

**Comment by juliodialpad** [comment]
Created the external account, and changed to a `connection_url` to avoid exposing servers, port, users, etc.

cc: @nickfirespotter

**Comment by braddialpad** [comment]
Couple things I would like to do here to introduce this.

Send out email to our users notifying them that this email address is not spam 😆 
Write an initial blog post to make sure everything is working correctly

I will do this after this is merged.

This will not immediately send anything out after we merge right? only when a new commit with a blog post goes to production?

**Comment by juliodialpad** [comment]
> Couple things I would like to do here to introduce this.
> 
> Send out email to our users notifying them that this email address is not spam 😆 Write an initial blog post to make sure everything is working correctly
> 
> I will do this after this is merged.
> 
> This will not immediately send anything out after we merge right? only when a new commit with a blog post goes to production?

Right, this will only send communication when the posts reach production

**Review by iropolo — APPROVED** [review]
Thanks, lets move forward with this one.

Let's wait for required approvals for this if it is necessary.

File: .github/workflows/send-email.yml
Comment (CoreyFirespotter) [inline .github/workflows/send-email.yml:88]: Did you mean to BCC the From email? Can probably be removed.

File: .github/workflows/send-email.yml
Comment (CoreyFirespotter) [inline .github/workflows/send-email.yml:83]: I would suggest specifying server_address, server_port, username and password parameters separately instead of using connection_url. It’s more debuggable / discoverable if the sender address is in the clear ( `dialtone.updates@gmail.com`) along with the Gmail server address + port. The password can still be a GMail app password stored in the GitHub repo’s secrets for security.

File: .github/workflows/send-email.yml
Comment (CoreyFirespotter) [inline .github/workflows/send-email.yml:89]: Note, Some receivers may reject this cross-site behavior. If it works for us, great. If not, consider sending the email to dialtone@dialpad.com with each recipient on BCC. Or keeping it simple and sending from dialtone.updates@gmail.com to dialtone-users@dialapad.com.

File: .github/workflows/send-email.yml
Comment (juliodialpad) [reply .github/workflows/send-email.yml:88]: I put it so we can have a copy of the emails sent on the inbox of the external account. I'd prefer to keep it as the emails are not on the sent box.

File: .github/workflows/send-email.yml
Comment (juliodialpad) [reply .github/workflows/send-email.yml:83]: I changed to this for security reasons, the action still has a debug option enabled that would help us in the case that the emails are not working and we need to debug.

It's better to not having exposed the smtp server, port and username. All of those parameters are set on the MAIL_CONECTION secret along with the app password.

If preferred I can change it, it's just my opinion that this is more secure and doesn't affect how debuggable this is.

File: .github/workflows/send-email.yml
Comment (juliodialpad) [reply .github/workflows/send-email.yml:89]: I don't understand completely what's the suggestion here.

We're currently sending from dialtone.updates@gmail.com to dialtone-users@dialpad.com.

The reply_to: dialtone@dialpad.com is so users can respond to an email all dialtone members can see, otherwise the reply will end up on an inbox that not many people can see right now.

You're suggesting to change that reply_to?

File: .github/workflows/send-email.yml
Comment (braddialpad) [inline .github/workflows/send-email.yml]: Let's change the subject to `[dialtone] New blog post: ${{ env.BLOG_POST_HEADING }}` Since this seems to be the standard way of categorizing emails at dialpad. Makes it easier to add to filters for it.

**Comment by braddialpad** [comment]
We're going to merge this in. @CoreyFirespotter Let us know if you have any remaining concerns with this after @juliodialpad's responses

**Pull request merged**
